lib/Dialect/Secret/IR/SecretPatterns.cpp

```diff
@@ -43,6 +43,29 @@ LogicalResult RemoveUnusedGenericArgs::matchAndRewrite(
   return hasUnusedOps ? success() : failure();
 }
 
+LogicalResult RemoveUnusedYieldedValues::matchAndRewrite(
+    GenericOp op, PatternRewriter &rewriter) const {
+  bool foundUnused = false;
```

Comment:
small nit: you can remove this bool and just check `!valuesToRemove.empty()`

Reply:
Fixed!